poncho/src/poncho/package_serverize.py

```diff
@@ -172,18 +167,29 @@ def pack_library_code(path, envpath):
 # Python's exec or Jupyter Notebooks won't work here.
 # @param functions  A list of functions to generate the hash value from.
 # @return           a string of hex characters resulted from hashing the contents and names of functions.
-def generate_functions_hash(functions: list) -> str:
+def generate_functions_hash(functions: list, import_modules=None) -> str:
+    import re
+    import sys
+
     source_code = ""
```

Comment:
A better implementation would be to have `source_code = []` as a list to which you `source.append(some_string)`. Then before computing the hash do a `" ".join(source_code)`.  Modules can added with `source_code.extend(["import " + module.__name___ etc.])`